### PR TITLE
feat: added alert rule empty state and product edu

### PR DESCRIPTION
**Pull request by SagarRajput-7**
#### Screenshots

https://github.com/SigNoz/signoz/assets/162284829/190c6bbe-6eea-4fd7-a8a8-ab19c019410f

Screenshot 2024-06-11 at 11 03 42 AM

### CODE REVIEW

**Review by makeavish — APPROVED** [review]

File: frontend/src/container/ListAlertRules/AlertsEmptyState/AlertsEmptyState.tsx [hunk 1]
```diff
@@ -0,0 +1,233 @@
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+/* eslint-disable jsx-a11y/img-redundant-alt */
```
Comment (YounixM) [inline frontend/src/container/ListAlertRules/AlertsEmptyState/AlertsEmptyState.tsx:4]: Why are we disabling the function return type and img redundant alt ?

File: frontend/src/container/ListAlertRules/AlertsEmptyState/AlertsEmptyState.tsx [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+								link: 'https://youtu.be/xjxNIqiv4_M',
+								leftIconVisible: true,
+								rightIconVisible: true,
+							})}
+						</div>
+
+						{InfoLinkText({
+							infoText: 'How to create Metrics-based alerts',
+							link:
+								'https://signoz.io/docs/alerts-management/metrics-based-alerts/?utm_source=product&utm_medium=alert-empty-page',
```
Comment (YounixM) [inline frontend/src/container/ListAlertRules/AlertsEmptyState/AlertsEmptyState.tsx:121]: Maintain constants so that it's easy to read.

Comment (YounixM) [reply frontend/src/container/ListAlertRules/AlertsEmptyState/AlertsEmptyState.tsx:121]: It's all for the infoText and Links

Comment (SagarRajput-7) [reply frontend/src/container/ListAlertRules/AlertsEmptyState/AlertsEmptyState.tsx]: done moved to separate util and files

File: frontend/src/container/ListAlertRules/AlertsEmptyState/AlertsEmptyState.tsx (hunk 1 above)
Comment (SagarRajput-7) [reply frontend/src/container/ListAlertRules/AlertsEmptyState/AlertsEmptyState.tsx]: fixed it

File: frontend/src/container/ListAlertRules/AlertsEmptyState/alertLinks.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+		rightIconVisible: true,
+	},
+	{
+		infoText: 'How to create Trace-based alerts',
+		link:
+			'https://signoz.io/docs/alerts-management/trace-based-alerts/?utm_source=product&utm_medium=alert-empty-page',
+		leftIconVisible: false,
+		rightIconVisible: true,
+	},
+];
```
Comment (SagarRajput-7) [inline frontend/src/container/ListAlertRules/AlertsEmptyState/alertLinks.ts:23]: @YounixM - I have not moved/created constant here for these strings has these wont be used in any case more than one time.

**Comment by makeavish** [comment]
Missing facing issues button?

Screenshot 2024-06-14 at 16 30 36

**Comment by SagarRajput-7** [comment]
> Missing facing issues button?
> 
> Screenshot 2024-06-14 at 16 30 36
> 

Are you checking in an account where it's cloud and have intercom

**Comment by makeavish** [comment]
> > Missing facing issues button?
> > Screenshot 2024-06-14 at 16 30 36
> 
> Are you checking in an account where it's cloud and have intercom

I am checking where I have isCloud as true. Intercom is not shown though. Earlier facing issues button were visible even without intercom. Was there any change recently to only show facing issues button when intercom is visible? If yes, it's fine.

**Comment by SagarRajput-7** [comment]
> > > Missing facing issues button?
> > > Screenshot 2024-06-14 at 16 30 36
> > 
> > 
> > Are you checking in an account where it's cloud and have intercom
> 
> I am checking where I have isCloud as true. Intercom is not shown though. Earlier facing issues button were visible even without intercom. Was there any change recently to only show facing issues button when intercom is visible? If yes, it's fine.

no dependency on intercom is necessary, because what will happen after we click if env doesn't have intercom

**Review by makeavish — APPROVED** [review]

File: frontend/src/container/ListAlertRules/AlertsEmptyState/alertLinks.ts (hunk 3 above)
Comment (YounixM) [reply frontend/src/container/ListAlertRules/AlertsEmptyState/alertLinks.ts:23]: It helps with code readability too.

**Review by YounixM — APPROVED** [review]

**Pull request merged**